tools/project_manager/diff.php

```diff
@@ -135,17 +138,44 @@
         $_GET["format"] = "remove";
     }
     $format_url = "?" . attr_safe(http_build_query($_GET));
+    $format_label = html_safe($format_label);
     echo "| <a href='$format_url'>$format_label</a>\n";
 }
 
+if (user_can_mentor_in_any_round()) {
+    // show option to change diff style
+    if ($bb_diffs) {
+        $bb_diffs_label = _("Show table-style diff");
+        $_GET["bb_diffs"] = "off";
+    } else {
+        $bb_diffs_label = _("Show BBCode-style diff");
+        $_GET["bb_diffs"] = "on";
+    }
+    $bb_diffs_url = "?" . attr_safe(http_build_query($_GET));
+    $bb_diffs_label = html_safe($bb_diffs_label);
+    echo "| <a href='$bb_diffs_url'>$bb_diffs_label</a>\n";
+}
+
+$diffurl = "$code_url/tools/project_manager/diff.php?project=$projectid&image=$image&L_round_num=$L_round_num&R_round_num=$R_round_num";
+
+// Provide button to copy BBCode diffs not wrapped in HTML
+if ($bb_diffs) {
+    $diffEngineCopy = new DifferenceEngineWrapperBB($diffurl);
+    $diffEngineCopy->setText($L_text, $R_text);
+    $cliptext = javascript_safe($diffEngineCopy->getDiff($L_label, $R_label));
```

Comment:
Instead of passing the text directly, maybe just have a function that reads the diff from the DOM so you don't need to encode at all.

Reply:
If we're showing the bbcode, I meant grab it from the < pre > tag in the DOM maybe in the function like
```js
navigator.clipboard.writeText(document.getElementById('#preTag').textcontent).catch(function () {
  alert("Unable to copy!");
});
```

Reply:
Thanks for the code snippet - I agree that if we're showing the unmodified bbcode and also not putting the button on the table-style diffs page, that is much neater. From initial user feedback, it may be that the Copy everything to Clipboard button is not really useful for what they do, which is to copy diffs individually into a message template. I'll solicit more feedback after changing/fixing a couple of things, to help make that decision.